tools/reviewer/job/sync/GitHubReader.java

```diff
@@ -0,0 +1,257 @@
+                .filter(comment -> comment.getPosition() == position)
+                .collect(Collectors.toList()));
+
+    Thread.Builder thread = Thread.newBuilder();
+    positionComments.forEach(
+        comment ->
+            thread
+                .setRepoId(pr.getTitle().replace("D", ""))
+                .setCommitId(setThreadCommitId(pr, comment))
+                .setFile(
+                    Protos.File.newBuilder()
+                        .setFilename(comment.getPath())
+                        .setRepoId(pr.getTitle().replace("D", ""))
+                        .setFilenameWithRepo(pr.getHead().getRepo().getFullName())
+                        .setCommitId(comment.getCommitId())
+                        // TODO: Think over how to save already received user's email from the
+                        // previous iteration. It can reduce the number of requests.
+                        .setUser(
+                            gitHubClient
+                                .getUser(
+                                    UserRequest.newBuilder()
+                                        .setLogin(pr.getUser().getLogin())
+                                        .build())
+                                .getUser()
+                                .getEmail())
+                        .build())
+                .setLineNumber(
+                    getLineNumber(
+                        getDiffPatchStrByFilename(pullRequestFiles, comment.getPath()),
+                        comment.getPosition(),
+                        getCommentSide(comment.getDiffHunk())))
+                .addComment(
+                    Comment.newBuilder()
+                        .setContent(comment.getBody())
+                        .setTimestamp(Instant.parse(comment.getUpdatedAt()).getEpochSecond())
+                        .setCreatedBy(
+                            // TODO: Email can be null if the user hides email. Think over how to
+                            // resolve this.
+                            gitHubClient
+                                .getUser(
+                                    UserRequest.newBuilder()
+                                        .setLogin(comment.getUser().getLogin())
+                                        .build())
+                                .getUser()
+                                .getEmail())
+                        .build())
+                .setType(Thread.Type.CODE));
+    return thread.build();
+  }
+
+  private LineNumberConverter.Side getCommentSide(String commentDiffHunk) {
+    List<String> newLinesSymbols = new ArrayList<>();
+    Pattern pattern = Pattern.compile(NEW_LINES_PATTERN);
+    Matcher matcher = pattern.matcher(commentDiffHunk);
+    while (matcher.find()) {
+      newLinesSymbols.add(matcher.group());
+    }
+    String lastNewlineSymbol = newLinesSymbols.get(newLinesSymbols.size() - 1);
+    // If the last newline symbol in comment diff hunk is `\n+`, it means the comment was left to
```

Comment:
left as in leave?
It's a little confusing with left/right.
Perhaps s/was left to/is on ?
Does it have the meaning you intended?

Reply:
Changed it.